uint32 command_id
